testscommon/txcachemocks/txGasHandlerMock.go

```diff
@@ -1,56 +1,84 @@
 package txcachemocks
 
 import (
+	"math/big"
+
+	"github.com/multiversx/mx-chain-core-go/core"
 	"github.com/multiversx/mx-chain-core-go/data"
 )
 
 // TxGasHandler -
 type TxGasHandler interface {
-	SplitTxGasInCategories(tx data.TransactionWithFeeHandler) (uint64, uint64)
-	GasPriceForProcessing(tx data.TransactionWithFeeHandler) uint64
-	GasPriceForMove(tx data.TransactionWithFeeHandler) uint64
 	MinGasPrice() uint64
-	MinGasLimit() uint64
-	MinGasPriceForProcessing() uint64
+	MaxGasLimitPerTx() uint64
+	ComputeTxFee(tx data.TransactionWithFeeHandler) *big.Int
 	IsInterfaceNil() bool
 }
 
 // TxGasHandlerMock -
 type TxGasHandlerMock struct {
-	MinimumGasMove       uint64
-	MinimumGasPrice      uint64
-	GasProcessingDivisor uint64
+	minGasLimit      uint64
+	minGasPrice      uint64
+	maxGasLimitPerTx uint64
+	gasPerDataByte   uint64
+	gasPriceModifier float64
 }
 
-// SplitTxGasInCategories -
-func (ghm *TxGasHandlerMock) SplitTxGasInCategories(tx data.TransactionWithFeeHandler) (uint64, uint64) {
-	moveGas := ghm.MinimumGasMove
-	return moveGas, tx.GetGasLimit() - moveGas
+// NewTxGasHandlerMock -
+func NewTxGasHandlerMock() *TxGasHandlerMock {
+	return &TxGasHandlerMock{
+		minGasLimit:      50000,
+		minGasPrice:      1000000000,
+		maxGasLimitPerTx: 600000000,
+		gasPerDataByte:   1500,
+		gasPriceModifier: 0.01,
+	}
 }
 
-// GasPriceForProcessing -
-func (ghm *TxGasHandlerMock) GasPriceForProcessing(tx data.TransactionWithFeeHandler) uint64 {
-	return tx.GetGasPrice() / ghm.GasProcessingDivisor
+// WithMinGasLimit -
+func (ghm *TxGasHandlerMock) WithMinGasLimit(minGasLimit uint64) *TxGasHandlerMock {
+	ghm.minGasLimit = minGasLimit
+	return ghm
 }
 
-// GasPriceForMove -
-func (ghm *TxGasHandlerMock) GasPriceForMove(tx data.TransactionWithFeeHandler) uint64 {
-	return tx.GetGasPrice()
+// WithMinGasPrice -
+func (ghm *TxGasHandlerMock) WithMinGasPrice(minGasPrice uint64) *TxGasHandlerMock {
+	ghm.minGasPrice = minGasPrice
+	return ghm
 }
 
 // MinGasPrice -
 func (ghm *TxGasHandlerMock) MinGasPrice() uint64 {
-	return ghm.MinimumGasPrice
+	return ghm.minGasPrice
+}
+
+// WithGasPriceModifier -
+func (ghm *TxGasHandlerMock) WithGasPriceModifier(gasPriceModifier float64) *TxGasHandlerMock {
+	ghm.gasPriceModifier = gasPriceModifier
+	return ghm
 }
 
-// MinGasLimit -
-func (ghm *TxGasHandlerMock) MinGasLimit() uint64 {
-	return ghm.MinimumGasMove
+// MaxGasLimitPerTx -
+func (ghm *TxGasHandlerMock) MaxGasLimitPerTx() uint64 {
+	return ghm.maxGasLimitPerTx
 }
 
-// MinGasPriceForProcessing -
-func (ghm *TxGasHandlerMock) MinGasPriceForProcessing() uint64 {
-	return ghm.MinimumGasPrice / ghm.GasProcessingDivisor
+// ComputeTxFee -
+func (ghm *TxGasHandlerMock) ComputeTxFee(tx data.TransactionWithFeeHandler) *big.Int {
```

Comment:
Generally speaking, we don't want to import the mock package from "mx-chain-storage-go". Therefore, we redefine this mock here, as well.